rust/src/datatypes/schema.rs

```diff
@@ -68,6 +68,18 @@ impl Schema {
         })
     }
 
+    pub(crate) fn validate(&self) -> Result<bool> {
```

Comment:
Could add a comment / doc somewhere to mention what is valid schema / field? Can do it in a follow up PR